doc/api/esm.md

```diff
@@ -1030,58 +997,61 @@ const baseURL = pathToFileURL(`${cwd()}/`).href;
 // CoffeeScript files end in .coffee, .litcoffee or .coffee.md.
 const extensionsRegex = /\.coffee$|\.litcoffee$|\.coffee\.md$/;
 
-export async function resolve(specifier, context, defaultResolve) {
+export async function resolve(specifier, context, next) {
+  // When it's not CoffeeScript, defer to another resolve hook.
+  if (extensionsRegex.test(specifier)) return next(specifier, context);
+
```

Comment:
Because most hooks are written with the call to `next` as the fallback/default condition. Like if it were a `switch`, the call to `next` goes at the end in `default:`.

When there’s only one condition being evaluated, it feels arbitrary; but imagine of there were two or more conditions here, like “if the specifier ends in a CoffeeScript extension” _or_ “if the parentUrl ends in a CoffeeScript extension” _or_ “if the `format` returned by a preceding hook is `coffee`“ etc. You wouldn’t want to put all of these in the condition at top in the negative to call `next` on, and then branch on each of these conditions _again_ in the body of the hook. It’s less duplicative to check for each one in turn and then call `next` at the end if no condition evaluates as true, like the structure of a `switch`.

Reply:
Not trying to play contrarian, but those don't sound like abort conditions then? If you can simply abort early, do; if you can't, don't.

My thinking for these examples is to highlight how and what needs to happen, in the most grokkable way: when your hook does not handle something, you need to call `next()`. I think putting that at the bottom makes it easily overlooked: when reading an example, people are looking to see how the author achieved the thing, and they stop reading when they do.

Reply:
At the very least, rewriting the example to change its style seems beyond the scope of adding chaining. Can this rewrite be put into its own follow-up PR, and the examples in this PR be rewritten as minimally as necessary to support chaining? And we can debate what makes the most sense to most people in a PR devoted to that.